source/tools/acpiexec/aeinstall.c

```diff
@@ -343,6 +350,108 @@ AeInstallRegionHandlers (
     }
 }
 
+/*******************************************************************************
+ *
+ * FUNCTION:    AeInstallGedHandler
+ *
+ * PARAMETERS:  ACPI_WALK_NAMESPACE callback
+ *
+ * RETURN:      Status
+ *
+ * DESCRIPTION: Walk entire namespace, install a handler for every GED
+ *              device found.
+ *
+ ******************************************************************************/
+static ACPI_STATUS
+AeInstallGedHandler (
+    ACPI_HANDLE             ObjHandle,
+    UINT32                  Level,
+    void                    *Context,
+    void                    **ReturnValue)
+{
+
+	ACPI_BUFFER             ReturnBuffer;
+	ACPI_STATUS Status;
+	ACPI_RESOURCE *ResourceList;
+	ACPI_RESOURCE_EXTENDED_IRQ *extended_irq_rsc;
+    ACPI_NAMESPACE_NODE     *Node;
+	ACPI_NAMESPACE_NODE		*EvtMethodNode;
+
+	/* Obtain the Namespace Node of this GED object handle. */
+	Node = AcpiNsValidateHandle (ObjHandle);
+	if (!Node)
+	{
+		return (AE_BAD_PARAMETER);
+	}
+
+	/*
+	 * A GED device must have one _EVT method.
+	 * Obtain the _EVT method and store it in the global
+	 * GED register.
+	 */
+	Status = AcpiNsSearchOneScope (
+		*ACPI_CAST_PTR (ACPI_NAME, METHOD_NAME__EVT),
+		Node,
+		ACPI_TYPE_METHOD,
+		&EvtMethodNode
+	);
+	if (ACPI_FAILURE (Status))
+	{
+		AcpiOsPrintf ("Failed to obtain _EVT method for the GED device.\n");
+		return Status;
+	}
+
+	ReturnBuffer.Pointer = NULL;
+    ReturnBuffer.Length = ACPI_ALLOCATE_LOCAL_BUFFER;
+
+	Status = AcpiGetCurrentResources (ObjHandle, &ReturnBuffer);
+	if (ACPI_FAILURE (Status))
+	{
+		AcpiOsPrintf ("AcpiGetCurrentResources failed: %s\n",
+			AcpiFormatException (Status));
+		return Status;
+	}
+
+	/* Traverse the _CRS resource list */
+	ResourceList = ACPI_CAST_PTR (ACPI_RESOURCE, ReturnBuffer.Pointer);
+	while (ResourceList->Type != ACPI_RESOURCE_TYPE_END_TAG) {
+
+		switch (ResourceList->Type) {
+		 case ACPI_RESOURCE_TYPE_EXTENDED_IRQ: {
+
+			/*
+			 * Found an Interrupt resource. Link the interrupt resource
+			 * and the _EVT method of this GED device in the GED event handler list.
+			 */
+				ACPI_GED_HANDLER_INFO *GedHandler =
+				ACPI_ALLOCATE (sizeof (ACPI_SCI_HANDLER_INFO));
+			if (!GedHandler)
+			{
+				return_ACPI_STATUS (AE_NO_MEMORY);
```

Comment:
3 errors total
430:'_AcpiFunctionName': undeclared identifier - I think you first need to use ACPI_FUNCTION_ENTRY() at the beginning of the function 
    ACPI_FUNCTION_ENTRY(AeInstallGedHandler); (ln 380)
this error is part of although I am not 100% sure right now and just this might not fix this error here

438 and 441:'Interrupts':is not a member of 'acpi_resource_extended irq' - you need to use u.Interrupts[0] both places to get rid of this error

You also might want to do an include for rsmisc.c file

Also 2 warnings on line 430:
'function': 'const char*' differs in levels of indirection from 'int' - you can try type casting the return value
'AcpiUtStatusExit': different types for formal and actual parameter 2 - which is the function name that needs to be fixed too

Reply:
Thank you for the review,

- I've added the ACPI_FUNCTION_ENTRY() to the top of the AeInstallGedHandler function.
- I'm returning AE_NO_MEMORY; if ACPI_ALLOCATE fails (I believe this solves the 2 additional warnings).

So far I have tested the build with GCC.
I will build this branch with MSVC too and confirm that all errors and warnings are removed -- I'll comment once I've performed the verification with MSVC.